pkg/helpers/helpers_test.go

```diff
@@ -221,3 +222,75 @@ func Test_DecodeSnowflakeAccountIdentifier(t *testing.T) {
 		require.ErrorContains(t, err, fmt.Sprintf("unable to read identifier: %s", id))
 	})
 }
+
+func TestParseRootLocation(t *testing.T) {
+	tests := []struct {
+		name     string
+		location string
+		id       string
+		path     string
+		wantErr  bool
+	}{
+		{
+			name:     "unquoted",
+			location: `@a.b.c`,
+			id:       `"a"."b"."c"`,
+		},
+		{
+			name:     "unquoted with path",
+			location: `@a.b.c/foo`,
+			id:       `"a"."b"."c"`,
+			path:     `foo`,
+		},
+		{
+			name:     "partially quoted",
+			location: `@"a".b.c`,
+			id:       `"a"."b"."c"`,
+		},
+		{
+			name:     "partially quoted with path",
+			location: `@"a".b.c/foo`,
+			id:       `"a"."b"."c"`,
+			path:     `foo`,
+		},
+		{
+			name:     "quoted",
+			location: `@"a"."b"."c"`,
+			id:       `"a"."b"."c"`,
+		},
+		{
+			name:     "quoted with path",
+			location: `@"a"."b"."c"/foo`,
+			id:       `"a"."b"."c"`,
+			path:     `foo`,
+		},
+		{
+			name:     "unquoted with path with dots",
+			location: `@a.b.c/foo.d`,
+			id:       `"a"."b"."c"`,
+			path:     `foo.d`,
+		},
+		{
+			name:     "quoted with path with dots",
+			location: `@"a"."b"."c"/foo.d`,
```

Comment:
maybe add tests with more slashes, there is a logic around them

Reply:
Done.